### ISSUE TITLE
Collabora not working anymore since NC26 (ARM)

### ISSUE THREAD

**Issue by Bobmorton-TX** [nextcloud-snap/nextcloud-snap#2477]
#### Describe the bug
Since the update to NC26 Collabora does not work anymore.
This seems to be due to the forced update of Collabora to version 23.5.104. There is a stale bug report that this version does not work by @Pilzinsel64  https://github.com/CollaboraOnline/richdocumentscode/issues/226

I managed to install the old working version 22.5.1301 and it was running ok for a while.
I suffered with high CPU load spikes after a few hours (https://github.com/nextcloud-snap/nextcloud-snap/issues/2457) and solved these by restarting the snap every 6 hours and replacing the Collabora version via script to 22.5.1301.

Since today Collabora wont start anymore. I have rebooted the device (Rock5), restarted NC snap. deactivated Collabora.
The 

Error `GuzzleHttp\Exception\ServerException: Server error: `GET https://cloud.sessionvictim.com/extra-apps/richdocumentscode_arm64/proxy.php?req=/hosting/capabilities` resulted in a `500 Internal Server Error` response`

#### To Reproduce
open office document.
blank screen - office not loading. no error message

#### Expected behavior
Open Office document

#### OS/snapd/snap version
Armbian 23.02.2 Jammy with Linux 5.10.110-rockchip-rk3588

Name Version Rev Tracking Publisher Notes
nextcloud 26.0.3snap1 35883 latest/stable nextcloud✓ -

snap 2.59.5
snapd 2.59.5
series 16
ubuntu 22.04

#### Logs
Error message when Collarbora app is enabled.

`GuzzleHttp\Exception\ServerException: Server error: `GET https://cloud.sessionvictim.com/extra-apps/richdocumentscode_arm64/proxy.php?req=/hosting/capabilities` resulted in a `500 Internal Server Error` response

[bug report snap 20230726.txt](https://github.com/nextcloud-snap/nextcloud-snap/files/12172676/bug.report.snap.20230726.txt)

**Comment by Bobmorton-TX** [comment]
would it be possible to revert to snap NC25 without losing data?

**Comment by Pilzinsel64** [comment]
> would it be possible to revert to snap NC25 without losing data?

Not really. If you updated previously, then you can use the command `sudo snap revert nextcloud`. But this works is only if you updated previously, so your last update MUST be from NC25 to NC26 and you didn't add many new data since then. I don't recommend you to go back, especially as I don't think you made manual backups.

But the workaround with the script that downgrades the app should do the trick. The other issues (high cpu usage, error after few days) I haven't had.

**Comment by Bobmorton-TX** [comment]
I have daily database and config backups via nextcloud.export -  and rsync backup of all data. 

can I setup a new snap instance (NC25) with that?

I cannot do the revert trick anymore. As soon as the nextcloud office app is activated I get a stream of these error messages:
`GuzzleHttp\Exception\ServerException: Server error: `GET https://cloud.sessionvictim.com/extra-apps/richdocumentscode_arm64/proxy.php?req=/hosting/capabilities` resulted in a `500 Internal Server Error` response`

I cannot change the Coder Server settings via the web admin interface - says server is unreachable. It seems the whole office to collabora connection is messed up. 

any idea how I can troubleshoot that?

**Comment by Pilzinsel64** [comment]
> I have daily database and config backups via nextcloud.export - and rsync backup of all data.

This is brilliant! See [the wiki entry](https://github.com/nextcloud-snap/nextcloud-snap/wiki/Export-and-Import) for export & import how to handle that. With this you should be able to setup a fresh install of nc25 and import your last v25 backup.

> any idea how I can troubleshoot that?

Not really, I absolutely have no idea, sorry. 😢
You may try your luck at https://help.nextcloud.com or create an issue at the corresponding GitHub repo (you can test if the Demo Servers does work).

**Comment by Bobmorton-TX** [comment]
I have setup a Collabora docker and this works with NC (i have to login to the admin interface of collabora once after container restart, otherwise it does not work)

searching for files in NC on the webfront end causes immediate reboot of the whole server. I fear my snap install is broken.
I tested Collarbora  23.5.104. with the current official NC docker image (NC27) and this works fine on my platform. 

could a reinstall of the snap help (NC26 and import of database and data)?

**Comment by Pilzinsel64** [comment]
You can give it a try, of course! But if it still not work, you should try purging the collabora apps completely by uninstalling them, Remove the app data and remove the tables from the database. Then you can pefform a fresh install of collabora.

**Comment by Bobmorton-TX** [comment]
I ended up installing a fresh snap install (26.0.1) and restoring the db and data. 
Using a script to replace Collabora Online - Built-in CODE Server (ARM64) with version 22.5.1301. Restarting NC snap every 6 hours because it the cpu load goes up significantly after a few hours. After restart back to very low level for some time.

I have tested NC26 docker with latest Collabora Online - Built-in CODE Server (ARM64)  and this works well.
So this must be an issue with the snap, i think.

**Comment by Pilzinsel64** [comment]
I take the liberty of closing the issue here as won't fix / not planned. The main issue is up-stream and built-in code is not officially supported in Nextcloud Snap.
Anyway I continue to provide support to the upstream maintainers to get around the snap enviroment (at least as far as I have abilities to) to help them getting the app compatible with Nextcloud snap.

**Comment by Pilzinsel64** [comment]
Everyone who is still reading this:
See https://github.com/CollaboraOnline/richdocumentscode/issues/226#issuecomment-1750198405

**Comment by scubamuc** [comment]
See working configuration here: https://github.com/nextcloud-snap/nextcloud-snap/wiki/Configure-CODE-and-Nextcloud-office-for-Nextcloud-snap

**Comment by Bobmorton-TX** [comment]
> See working configuration here: https://github.com/nextcloud-snap/nextcloud-snap/wiki/Configure-CODE-and-Nextcloud-office-for-Nextcloud-snap

I ran the steps exactly like listed this is not working for my Rock5 (arm) armbian setup.

**Comment by Pilzinsel64** [comment]
For arm devices there is a actual bug upstream. Issue linked. The steps has been tested on an amd64 device.
